### PR TITLE
Added collections.product

### PR DESCRIPTION
**Pull request by lfrancioli**

### CODE REVIEW

**Comment by lfrancioli** [comment]
Seems OK:
```
In [9]: print(hc.eval_expr('NA:Array[Int].product'))
None
```

**Comment by lfrancioli** [comment]
Seems OK:
```
In [9]: print(hc.eval_expr('NA:Array[Int].product'))
None
In [10]: print(hc.eval_expr('[1,2,3,4,NA:Int].product'))
24
```

I'm guessing the magic happens in `registerUnaryNAFilteredCollectionMethod`

**Comment by lfrancioli** [comment]
Actually, I realized I need this for aggregators... will update PR. Sorry.

**Comment by tpoterba** [comment]
ah, I missed the `unaryNAFilteredCollection` stuff.

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 1]
```diff
@@ -1890,6 +1890,15 @@ object FunctionRegistry {
     "Compute the sum by index. All elements in the aggregable must have the same length."
   )(aggregableHr(arrayHr(doubleHr)), arrayHr(doubleHr))
 
+  val productAggDocstring = """Compute the product of all non-missing elements. The empty product is one."""
+  registerAggregator[Int, Long]("product", () => new IntProductAggregator(), productAggDocstring)(aggregableHr(intHr), longHr)
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: You should just register one for longs and one for doubles. The implicit conversions will do the rest. This also means we don't need the IntProductAgg and FloatProductAgg stuff.

**Comment by lfrancioli** [comment]
Thanks Tim, hadn't realized that there would be an implicit conversion when evaluating the expressions!

**Pull request merged**